sklearn/linear_model/_glm/glm.py

```diff
@@ -79,19 +76,28 @@ class GeneralizedLinearRegressor(RegressorMixin, BaseEstimator):
         Specifies if a constant (a.k.a. bias or intercept) should be
         added to the linear predictor (X @ coef + intercept).
 
-    family : {'normal', 'poisson', 'gamma', 'inverse-gaussian'} \
-            or an ExponentialDispersionModel instance, default='normal'
-        The distributional assumption of the GLM, i.e. which distribution from
-        the EDM, specifies the loss function to be minimized.
+    base_loss_class : subclass of BaseLoss, default=HalfSquaredError
+        A `base_loss_class` contains a specific loss function as well as the link
+        function. The loss to be minimized specifies the distributional assumption of
+        the GLM, i.e. the distribution from the EDM. Here are some examples:
+
+        =======================  ========  ==========================
+        base_loss_class          Link       Target Domain
+        =======================  ========  ==========================
+        HalfSquaredError         identity  y any real number
+        HalfPoissonLoss          log       0 <= y
+        HalfGammaLoss            log       0 < y
+        HalfInverseGaussianLoss  log       0 < y
+        HalfTweedieLoss          log       dependend on tweedie power
+        =======================  ========  ==========================
 
-    link : {'auto', 'identity', 'log'} or an instance of class BaseLink, \
-            default='auto'
         The link function of the GLM, i.e. mapping from linear predictor
-        `X @ coeff + intercept` to prediction `y_pred`. Option 'auto' sets
-        the link depending on the chosen family as follows:
+        `X @ coeff + intercept` to prediction `y_pred`. For instance, with a log link,
+        we have `y_pred = exp(X @ coeff + intercept)`.
 
-        - 'identity' for Normal distribution
-        - 'log' for Poisson,  Gamma and Inverse Gaussian distributions
+    base_loss_params : dictionary, default={}
```

Comment:
If `base_loss_class` is immediately instantiated with `base_loss_params`, would it be simpler to take a unique parameter `base_loss=base_loss_class(**base_loss_params)` ?

Reply:
See comment below, https://github.com/scikit-learn/scikit-learn/pull/22548#discussion_r811550321.